pkg/cachedinstances/k8s/k8s_test.go

```diff
@@ -241,64 +229,24 @@ func TestIPToTags(t *testing.T) {
 }
 
 func TestNoHostNetworkPodsCached(t *testing.T) {
+	t.Parallel()
+
 	setupTest(t, func(t *testing.T, fixtures *testFixture) {
 		hostNetworkPod := pod()
 		hostNetworkPod.Status.HostIP = ipAddr
 		hostNetworkPod.Spec.HostNetwork = true
 		fixtures.podsWatch.Add(hostNetworkPod)
 		waitForFullCache(t, fixtures, 1)
 
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		instanceData, err := fixtures.cloudProvider.Instance(ctx, ipAddr)
-		require.NoError(t, err)
-
-		instance := instanceData[ipAddr]
+		instance, cacheHit := fixtures.provider.Peek(ipAddr)
+		require.True(t, cacheHit)
 		require.Nil(t, instance)
 	}, viper.New(), nodeName)
 }
 
-func TestWatchNodeOnly(t *testing.T) {
```

Comment:
Watch/list filtering happens on server so it's not really possible to test it using fakes. See below.
Deleted the test.